src/panoptes/pocs/focuser/astromechanics.py

```diff
@@ -0,0 +1,266 @@
+    that class' documentation for a complete list.
+    """
+
+    # Class variable to cache the device node scanning results
+    _adaptor_nodes = None
+
+    # Class variable to store the device nodes already in use. Prevents scanning known focuser adaptors and
+    # acts as a check against those ones assigned to incorrect ports.
+    _assigned_nodes = []
+
+    def __init__(self,
+                 name='Astromechanics Focuser Controller',
+                 model='astromechanics',
+                 initial_position=None,
+                 dev_node_pattern='/dev/tty.usbserial-00*',
+                 *args, **kwargs):
+        super().__init__(name=name, model=model, *args, **kwargs)
+        self.logger.debug('Initialising Astromechanics Lens Controller')
+
+        # Check that this node hasn't already been assigned to another adaptor
+        if self.port in Focuser._assigned_nodes:
+            message = 'Device node {} already in use!'.format(self.port)
+            self.logger.error(message)
+            warn(message)
+            return
+
+        try:
+            self.connect(self.port)
+        except (serial.SerialException,
+                serial.SerialTimeoutException,
+                AssertionError) as err:
+            message = 'Error connecting to {} on {}: {}'.format(self.name, self.port, err)
+            self.logger.error(message)
+            warn(message)
+            return
+
+        Focuser._assigned_nodes.append(self.port)
+        self._is_moving = False
+        self._initialise()
+
+    def __del__(self):
+        with suppress(AttributeError):
+            device_node = self.port
+            Focuser._assigned_nodes.remove(device_node)
+            self.logger.debug('Removed {} from assigned nodes list'.fomat(device_node))
+        with suppress(AttributeError):
+            self._serial_port.close()
+            self.logger.debug('Closed serial port {}'.format(self._port))
+
+    ##################################################################################################
+    # Properties
+    ##################################################################################################
+
+    @property
+    def is_connected(self):
+        """
+        Checks status of serial port to determine if connected.
+        """
+        connected = False
+        if self._serial_port:
+            connected = self._serial_port.isOpen()
+        return connected
+
+    @AbstractFocuser.position.getter
+    def position(self):
+        """
+        Returns current focus position in the lens focus encoder units.
+        """
+        response = self._send_command('P#', response_length=1)[0].replace("#", "")
+        return response
+
+    @property
+    def min_position(self):
+        """
+        Returns position of close limit of focus travel, in encoder units.
+        """
+        return None
+
+    @property
+    def max_position(self):
```

Comment:
as `min_` and `max_` position don't actually return things, I think think it is best if you delete them, which I think will cause it to fall back on the base class and raise NotImplementedError: https://github.com/panoptes/POCS/blob/3f265607ce8d4427e017d6c11d3e3ebc606deea7/src/panoptes/pocs/focuser/focuser.py#L144

Reply:
They are marked with `@abstractmethod` in the base class, meaning you have to override them. Probably not the best design.

Reply:
@JAAlvarado-Montes @AnthonyHorton does this focuser wrap-around or something? Why no `min`/`max`? And if that's the case let's describe it in the class' docstring.

Reply:
Basically we *only* have three commands for the astro mechanics controller: get current position (P#), move to position XX (MXX#), and set aperture to XX (AXX#). There isn't any command to get the min/max position or additional information available for Birgers.

Reply:
@JAAlvarado-Montes what happens when you specify a very big or small focus number? What focus value does it return? You could implement a quick scan of the max/min ranges by specifying really big & small moves and record the effective limits.

Reply:
@JAAlvarado-Montes can you answer this question?